### PR TITLE
Add Linux Commands deprecation document

### PR DESCRIPTION
**Pull request by mbbx6spp**

### CODE REVIEW

**Comment by mbbx6spp** [comment]
See this for formatted/presented ASCIIDOC output: https://github.com/mbbx6spp/styleguides/blob/shell-command-deprecations/bash/COMMANDS.asciidoc

**Comment by mbbx6spp** [comment]
@polynomial what other use case examples would be useful for `ss` command?

**Comment by polynomial** [comment]
The only other one that is pretty common for me is -p, to show whats binding a port (netstat -anp | grep LISTEN replacement)

It's unclear why some of the examples have -r and -n, aren't these contradictory?

Include `nslookup` in deprecated list, provide examples for:
* host HOSTNAME
* dig -t ANY hostname

wan't to be fancy pants? Include dnstracer example for looking up the tree of authority without multiple commands.

File: bash/COMMANDS.asciidoc [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+----
+
+TODO: More examples?
+
+=== ip addr
+
+If you are still using `ifconfig` in your scripts for whatever reason today
+you should consult this GIF and meditate on it for some time:
+
+image::http://giphy.com/gifs/wWEDxkj0qzjLa["Ashamed"]
```
Comment (polynomial) [inline bash/COMMANDS.asciidoc]: This doesn't work as an image because they do some bullshit detection stuff and try to feed you a full page i think...

Comment (mbbx6spp) [inline bash/COMMANDS.asciidoc]: Ahh, thanks. I thought it was just GitHub's weird ASCIIDOC rendering.

**Comment by mbbx6spp** [comment]
I will add the `-p` case for netstat and explain the `-r` and `-n` parts better. It's mostly a personal preference but I usually don't want to see `:documentum` instead of `:10002` or on overloaded ports `*:wap-wsp` instead of `*:9200` for Elasticsearch, but I usually want to see the resolved hostnames instead of IPs. I will illustrate with examples.

**Comment by mbbx6spp** [comment]
Note: I just added sections on `ip neighbor` and `ip link` this morning as replacements for `arp` and other `ifconfig` usages.

File: bash/COMMANDS.asciidoc (hunk 1 above)
Comment (mbbx6spp) [inline bash/COMMANDS.asciidoc]: I just put a link now rather than attempting to embed an image.

**Comment by mbbx6spp** [comment]
@polynomial I think I addressed the `-p` use case and the explanation of the `-n` option and why you would use it over not using it in the new subsection titled 'Look ma, no pipes' https://github.com/mbbx6spp/styleguides/blob/shell-command-deprecations/bash/COMMANDS.asciidoc#-look-ma-no-pipes-

**Comment by mbbx6spp** [comment]
Adding `nslookup` and alternatives now. Thanks!

**Comment by mbbx6spp** [comment]
I added the `dig` section here: https://github.com/mbbx6spp/styleguides/blob/shell-command-deprecations/bash/COMMANDS.asciidoc#dig

Do we really need `dnstracer` which is non-standard command? (See last dig command example) :)

**Comment by polynomial** [comment]
OMG THIS IS ALL GREAT AND I LOVE IT SO MUCH.

**Comment by mbbx6spp** [comment]
WOW. SO CAPS, MUCH OMG.

**Comment by mbbx6spp** [comment]
As per @stephenweber I updated the structure of the COMMANDS document to show the outline more as it was confusing and added the 'Alternatives' level.

**Pull request merged**